webgpu-render-demo/script.js

```diff
@@ -0,0 +1,146 @@
+// Clear color for GPURenderPassDescriptor
+const clearColor = { r: 0.0, g: 0.5, b: 1.0, a: 1.0 };
+
+// Vertex data for triangle
+// Each vertex has 8 values representing position and color: X Y Z W R G B A
+
+const vertices = new Float32Array([
+  0.0,  0.6, 0, 1, 1, 0, 0, 1,
+ -0.5, -0.6, 0, 1, 0, 1, 0, 1,
+  0.5, -0.6, 0, 1, 0, 0, 1, 1
+]);
+
+// Vertex and fragment shaders
+
+const shaders = `
+struct VertexOut {
+  @builtin(position) position : vec4f,
+  @location(0) color : vec4f
+}
+
+@vertex
+fn vertex_main(@location(0) position: vec4f,
+               @location(1) color: vec4f) -> VertexOut
+{
+  var output : VertexOut;
+  output.position = position;
+  output.color = color;
+  return output;
+}
+
+@fragment
+fn fragment_main(fragData: VertexOut) -> @location(0) vec4f
+{
+  return fragData.color;
+}
+`;
+
+// Main function
+
+async function init() {
+  // 1: request adapter and device
+  if (!navigator.gpu) {
+    throw Error('WebGPU not supported.');
+  }
+
+  const adapter = await navigator.gpu.requestAdapter();
+  if (!adapter) {
+    throw Error('Couldn\'t request WebGPU adapter.');
+  }
+
+  const device = await adapter.requestDevice();
```

Comment:
**🎯 suggestion:** I think it might be helpful to rename `device` to `gpuDevice` for clarity.